src/helpers/openInGitpod.ts

```diff
@@ -0,0 +1,55 @@
+import { LocalStorage, open, showToast, Toast } from "@raycast/api";
+import { getPreferenceValues } from "@raycast/api";
+
+interface Preferences {
+  preferredEditor: string;
+  useLatest: boolean;
+  preferredEditorClass: "g1-standard" | "g1-large";
+}
+
+export default async function OpenInGitpod(contextUrl: string, type: "Branch" | "Pull Request" | "Issue" | "Repository", repository: string, context?: string) {
+  let preferences = getPreferenceValues<Preferences>();
+
+  if (type === "Branch" || type === "Pull Request" || type === "Issue") {
+    const item = await LocalStorage.getItem<string>(`${repository}%${context}`)
+    const contextPref = item ? await JSON.parse(item) : null
+    if (contextPref && contextPref.preferredEditor && contextPref.preferredEditorClass) {
+      preferences = contextPref
+    } else {
+      const repoItem = await LocalStorage.getItem<string>(`${repository}`);
+      const repoPref = repoItem ? await JSON.parse(repoItem) : null
+      if (repoPref && repoPref.preferredEditor && repoPref.preferredEditorClass) {
+        preferences = repoPref
+      }
+    }
+  } else if (type === "Repository") {
+    const item = await LocalStorage.getItem<string>(`${repository}`);
+    const repoPref = item ? await JSON.parse(item) : null
+    if (repoPref && repoPref.preferredEditor && repoPref.preferredEditorClass) {
+      preferences = repoPref
+    }
+  }
+
+  if (type === "Branch") {
+    //visit branch 
+  } else if (type === "Pull Request") {
+    //vitit pr
+  } else if (type === "Issue") {
+    //visit issue
+  }
+
+  try {
+    await showToast({
+      title: "Launching your workspace",
+      style: Toast.Style.Success,
+    });
+    setTimeout(() => {
+      open(`https://gitpod.io/?useLatest=${preferences.useLatest}&editor=${preferences.preferredEditor}&workspaceClass=${preferences.preferredEditorClass}#${contextUrl}`);
```

Comment:
suggestion: to fix a bug in which the latest editor option is not pre-selected, we can add `-latest` to the option to mitigate it.

Example:

`goland` with <kbd>Latest</kbd> will be `goland-latest` instead of stable `goland`.

AFAIK `useLatest` only controls whether the latest options are actually shown.

```suggestion
      open(`https://gitpod.io/?useLatest=${preferences.useLatest}&editor=${preferences.preferredEditor}${preferences.useLatest ? "-latest" : ""}&workspaceClass=${preferences.preferredEditorClass}#${contextUrl}`);
```

Reply:
Ohhh got it, I'll make the changes!

Also, what do you mean "useLatest" is used for showing latest options?😅.. which options are we talking about here....

Reply:
By showing latest options, I mean showing the options labelled as <kbd>Latest</kbd> in here: (I believe if you do `useLatest=false` it will not show them)
<img width="652" alt="image" src="https://user-images.githubusercontent.com/29888641/225602480-9ec2f517-1809-45f8-ad4b-cdb749cb52cf.png">
https://gitpod.io/new/#https://github.com/gitpod-io/gitpod

Reply:
Ohhh! I see, I suppose I can totally remove this from the `url` as we don't really direct the user to this screen (as I've removed the `showOptions` field, rather the workspace just opens up!